README.md

```diff
@@ -5,13 +5,104 @@ Group a list of signals into one hash table signal. The output signal will conta
 
 If *one_value* is True, the Signal attributes will be just a single matching value instead of a list of all matching values. If multiple matches, then the last signal processed will be the value used.
 
+If *group_by* is used, a Signal attribute (key) with the default (pre-assigned) name *group* will have the value of *group_by* in the output signal.
+
+**Example:**
+
+_Signals Input_:
+
+```python
+[
+{ "type": "important", "color": "red", "age": 32 },
+{ "type": "minor", "color": "red", "age": 9 },
+{ "type": "important", "color": "orange", "age": 14 },
+{ "type": "standard", "color": "red", "age": 1 }
+]
+```
+
+_Block Config with key based on "type"_:
+
+```python
+key : {{ $type }}
+value : {{ $age }}
+one_value: False
+```
+
+_Signal Output_:
+
+```python
+{
+    "important": [32, 14],
+    "standard": [1],
+    "minor": [9],
+    "group": ""
+}
+```
+_Block Config with key based on "color"_:
+
+```python
+key : {{ $color }}
+value : {{ $age }}
+one_value: False
+```
+
+_Signal Output_:
+
+```python
+{
+    "red": [32, 9, 1],
+    "orange": [14],
+}
+```
+
+_Block Config with key based on "type" and "One Value Per Key" checked (one_value = True)_:
+
+```python
+key : {{ $type }}
+value : {{ $age }}
+one_value: True
+```
+
+_Signal Output_:
+
+```python
+{
+    "important": [14],
+    "standard": [1],
+    "minor": [9]
+    "group": ""
+}
+```
+
+_Block Config using *group_by* to assign name to the *group* Signal attribute_:
+
+```python
+key : {{ $type }}
+value : {{ $age }}
+group_by: "priority"
```

Comment:
This is a somewhat obscure example and may be confusing. You are grouping by a literal string here, it may make more sense to group by one of the attributes since it will indicate how the block will spit out multiple signals.